### PR TITLE
Check profiler accuracy

### PR DESCRIPTION
**Pull request by jd**
## fix(profile): only create scheduler if exporters are present

If no exporters are configured, there is no need to create any scheduler to
export data.

This allows to keep the event in memory without having the scheduler resetting
the recorder every N seconds.

## tests: add accuracy test for wall and CPU time

This adds a pretty simple accuracy test for the stack collector. It checks that
the wall and CPU time metering is actually correct.

### CODE REVIEW

File: tests/profile/test_accuracy.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        if event.thread_name == "MainThread":
+            for idx, frame in enumerate(event.frames):
+                time_spent_ns[idx][frame[2]] += event.wall_time_ns
+                cpu_spent_ns[idx][frame[2]] += event.cpu_time_ns
+
+    assert {"spend_3", "spend_1", "spend_cpu_2"}.issubset(set(time_spent_ns[0].keys())), time_spent_ns
+    assert {"spend_4", "spend_7", "spend_16"}.issubset(set(time_spent_ns[1].keys())), time_spent_ns
+
+    assert almost_equal(total_time(time_spent_ns, "spend_3"), 9e9)
+    assert almost_equal(total_time(time_spent_ns, "spend_1"), 2e9)
```
Comment (Kyle-Verhoog) [inline tests/profile/test_accuracy.py]:
```suggestion
    assert almost_equal(total_time(time_spent_ns, "spend_1"), 1e9)
```
?

File: tests/profile/test_accuracy.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+
+def test_accuracy():
+    p = profiler.Profiler()
+    p.start()
+    spend_16()
+    p.stop(flush=False)
+    recorder = list(p.recorders)[0]
+    # First index is the stack position, second is the function name
+    time_spent_ns = collections.defaultdict(lambda: collections.defaultdict(lambda: 0))
```
Comment (Kyle-Verhoog) [inline tests/profile/test_accuracy.py]: cool!

File: tests/profile/test_accuracy.py (hunk 1 above)
Comment (jd) [reply tests/profile/test_accuracy.py]: The function is called twice, so its total time is 2 seconds.

**Comment by jd** [comment]
The goal is not to have them flaky and to make sure the profiler works fine, wherever run. :)

File: tests/profile/test_accuracy.py (hunk 1 above)
Comment (Kyle-Verhoog) [reply tests/profile/test_accuracy.py]: ah missed that!

**Pull request merged**